src/components/dialogs/ag-grid-table-rhf/custom-ag-grid-table.js

```diff
@@ -0,0 +1,308 @@
+    name,
+    columnDefs,
+    defaultRowData,
+    fromCsvDataToFormValues, // this used to transform rows from csv file, to data can be displayed in the table
+    csvFileHeaders,
+    csvInitialData, // this is used if csv file generated has some row by default (comments for example)
+    ...props
+}) => {
+    const classes = useStyles();
+    const theme = useTheme();
+    const [gridApi, setGridApi] = useState(null);
+    const [selectedRows, setSelectedRows] = useState([]);
+    const [uploaderOpen, setUploaderOpen] = useState(false);
+    const { control } = useFormContext();
+    const [upDisabled, setUpDisabled] = useState(false);
+    const [downDisabled, setDownDisabled] = useState(false);
+    const intl = useIntl();
+    const { getValues } = useFormContext();
+
+    const { fields, append, remove, move, update } = useFieldArray({
+        control,
+        name: name,
+    });
+
+    // we add the unique generated id from react hook form to each row, so we can use it in getRowId
+    // the unique id used to identify the selected rows, so we can keep the selection when row index is changed.
+    const defaultRowsWithId = useMemo(() => {
+        const rows = getValues(name);
+        if (!rows) {
+            return [];
+        }
+        return getValues(name).map((v, i) => {
+            return {
+                id: fields[i].id,
+                ...v,
+            };
+        });
+    }, [fields, getValues, name]);
+
+    const [rowData, setRowData] = useState(defaultRowsWithId);
+
+    const handleAddRow = () => {
+        append(defaultRowData);
+        setRowData((oldValues) => {
+            const newVal = {
+                id: fields[fields.length - 1].id,
+                ...defaultRowData,
+            };
+            return [...oldValues, newVal];
+        });
+    };
+
+    const handleDeleteRows = () => {
+        selectedRows.forEach((val, index) => {
+            setSelectedRows(gridApi.api.getSelectedRows());
+            const rows = getValues(name);
+            const idx = rows.findIndex((obj) => {
+                for (let key in val) {
+                    if (obj[key] !== val[key]) {
+                        return false;
+                    }
+                }
+                return true;
+            });
+            remove(idx);
+        });
+        setRowData(getValues(name));
+    };
+
+    const getIndex = (val) => {
+        return getValues(name).findIndex((row) => row.id === val.id);
+    };
+    const handleMoveRowUp = () => {
+        selectedRows.forEach((val, index) => {
+            const idx = getIndex(val);
+            move(idx, idx - 1);
+        });
+        setUpDisabled(selectedRows.some((r) => getIndex(r) === 0));
+        setDownDisabled(
+            selectedRows.some((r) => getIndex(r) === rowData.length - 1)
+        );
+        setRowData(getValues(name));
+    };
+
+    const handleMoveRowDown = () => {
+        selectedRows.forEach((val, index) => {
```

Comment:
Can't works, you have to iterate from the last one. Check the same method here  https://github.com/gridsuite/gridstudy-app/blob/fe947e88ac5f254166b62481aa19692cebe6c84e/src/components/utils/dnd-table/dnd-table.js#L295